R/paths.R

```diff
@@ -52,14 +52,13 @@ redoc_index <- function() {
 #'   redoc_spec("https://docs.docker.com/engine/api/v1.38.yaml",
 #'    redoc_options = list(scrollYOffset = 250, disableSearch = TRUE))
 #' }
-#' @importFrom jsonlite toJSON
 #' @export
 #' @rdname redoc_spec
 redoc_spec <- function(spec_url = "https://redocly.github.io/redoc/openapi.yaml",
                        redoc_options = structure(list(), names = character())) {
   index_file <- redoc_index()
   index_txt <- paste0(readLines(index_file), collapse = "\n")
   index_txt <- sub("https://redocly\\.github\\.io/redoc/openapi\\.yaml", spec_url, index_txt)
-  index_txt <- sub("\\{\\}", toJSON(redoc_options, auto_unbox = TRUE), index_txt)
+  index_txt <- sub("\\{\\}", jsonlite::toJSON(redoc_options, auto_unbox = TRUE), index_txt)
```

Comment:
For personal knowledge, why calling it directly instead of importing? I had the impression that using :: had an impact on execution time. Although it does not matter at all here.

Reply:
The execution time comment makes sense. I have not heard it, but I have not done too many things where it makes a difference.

I use `::` mainly for readability. In this package it doesn't really matter, but in other packages like `shiny` there is a local version of `toJSON()` which calls `jsonlite::toJSON()`.

https://github.com/rstudio/shiny/blob/8ec6275f9a49b5b6d5a307febaaf5bee8086c99a/R/shiny.R#L44-L58

It's more of a habit that allows me to scan the code without knowledge of where unknown / external functions are coming from. If I see a function without a namespace, then I can assume it is defined locally.

I believe I've had some issues with `@importFrom` when interacting with S3 methods. (This was not consistent.)

Reply:
Exceptions:

I gladly make the exception to this habit for infix operators like ``ggplot2::`+.gg` `` or the magrittr pipe (http://adv-r.had.co.nz/Functions.html#infix-functions)

In `GGally`, I import all of `ggplot2` as `GGally` is more of an extension package to `ggplot2`.

... so not a hard rule by any means.